express-postgres/server.ts

```diff
@@ -0,0 +1,110 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const app = express();
+
+pg.types.setTypeParser(pg.types.builtins.NUMERIC, parseFloat);
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+app.use((req, res, next) => {
+  console.log('The date is', new Date());
+  next();
+});
+
+app.get('/api/actors', async (req, res, next) => {
+  try {
+    const sql = `
+    select
+      "actorID",
+      "firstName",
+      "lastName"
+    from "actors";
+    `;
+    const result = await db.query(sql);
+    const actors = result.rows;
+    res.send(actors);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/films', async (req, res, next) => {
+  try {
+    const sql = `
+    select
+      "title",
+      "filmId",
+      "replacementCost"
+    from "films"
+    order by "replacementCost" desc
+    `;
+    const results = await db.query(sql);
+    const films = results;
+    if (!films) {
+      throw new ClientError(404, `films not found`);
```

Comment:
When returning all the items, we usually don't return a 404 but instead return an empty array. `results.rows` will be an empty array if there aren't any films. So you don't need this check when returning all items.